data/registry_model/interface.py

```diff
@@ -394,6 +401,14 @@ def get_cached_namespace_region_blacklist(self, model_cache, namespace_name):
         the list could not be loaded.
         """
 
+    @abstractmethod
+    def get_cached_repo_blob(self, model_cache, namespace_name, repo_name, blob_digest):
+        """
+        Returns the blob in the repository with the given digest if any or None if none.
+
+        Caches the result in the caching system.
+        """
```

Comment:
This function is relied upon by callers of the `OCIModel`, so it made sense to add it to the interface.